src/three-components/TextureUtils.ts

```diff
@@ -168,118 +165,253 @@ export default class TextureUtils extends EventDispatcher {
+      if (i == 0 && dir == 1)
+        continue;
+      vec3 sampleDirection = vWorldDirection;
+      float xz = length(sampleDirection.xz);
+      float weight = weights[i];
+      if (latitudinal) {
+        float diTheta = dTheta * float(dir * i) / xz;
+        mat2 R = mat2(cos(diTheta), sin(diTheta), -sin(diTheta), cos(diTheta));
+        sampleDirection.xz = R * sampleDirection.xz;
+        texColor += weight * RGBEToLinear(textureCube(tCube, sampleDirection));
       } else {
-        if (skybox != null) {
-          // Infer the environment from the skybox if we have one:
-          environmentMap = skybox.texture;
-        } else {
-          // Otherwise, no skybox URL was specified, so fall back to generating
-          // the environment:
-          // TODO(#336): can cache this per renderer and color
-          environmentMap = this.environmentMapGenerator.generate();
-          environmentMapWasGenerated = true;
-        }
+        float diTheta = dTheta * float(dir * i);
+        mat2 R = mat2(cos(diTheta), sin(diTheta), -sin(diTheta), cos(diTheta));
+        vec2 xzY = R * vec2(xz, sampleDirection.y);
+        sampleDirection.xz *= xzY.x / xz;
+        sampleDirection.y = xzY.y;
+        texColor += weight * RGBEToLinear(textureCube(tCube, sampleDirection));
       }
+    }
+  }
+  gl_FragColor = linearToOutputTexel(texColor);
+}
+      `,
+      side: BackSide,
+      depthTest: false,
+      depthWrite: false
+    });
 
-      if (options.pmrem) {
-        // Apply the PMREM pass to the environment, which produces a distinct
-        // texture from the source:
-        const nonPmremEnvironmentMap = environmentMap;
-
-        environmentMap = this.pmremPass(nonPmremEnvironmentMap);
-
-        // If the source was generated, then we should dispose of it right away
-        if (environmentMapWasGenerated) {
-          nonPmremEnvironmentMap.dispose();
-        }
-      } else if (environmentMapWasGenerated) {
-        (environmentMap as any).userData = {
-          ...userData,
-          ...({
-            mapping: 'Cube',
-          })
-        };
-      }
+    blurScene.add(new Mesh(geometry, blurMaterial));
 
-      return {environmentMap, skybox};
-    } catch (error) {
-      if (skybox != null) {
-        (skybox as any).dispose();
-      }
+    const blurCamera = new CubeCamera(0.1, 100, cubeResolution);
+    const tempTarget = blurCamera.renderTarget;
+    tempTarget.texture.type = UnsignedByteType;
+    tempTarget.texture.format = RGBEFormat;
+    tempTarget.texture.encoding = RGBEEncoding;
+    tempTarget.texture.magFilter = NearestFilter;
+    tempTarget.texture.minFilter = NearestFilter;
+    tempTarget.texture.generateMipmaps = false;
 
-      if (environmentMap != null) {
-        (environmentMap as any).dispose();
-      }
+    var gammaOutput = this.renderer.gammaOutput;
+    var toneMapping = this.renderer.toneMapping;
+    var toneMappingExposure = this.renderer.toneMappingExposure;
```

Comment:
In modern JS, there are few good reasons to use `var` ever. Use `const` or `let` instead (in this case probably `const`).